### PR TITLE
Decouple hook executors from `GitMacheteRepository*`

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: backend/api/src/main/java/com/virtuslab/gitmachete/backend/hooks/BaseHookExecutor.java [hunk 1]
```diff
@@ -1,8 +1,9 @@
-package com.virtuslab.gitmachete.backend.impl.hooks;
+package com.virtuslab.gitmachete.backend.hooks;
```
Comment (PawelLipski) [inline backend/api/src/main/java/com/virtuslab/gitmachete/backend/hooks/BaseHookExecutor.java:1]: This PR moves some (albeit not much) of the hooks implementation into `backend:api` (which shouldn't contain the implementation)... which I'm not very happy with, but I've considered the alternatives and it ended up being the best option currently

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java [hunk 2]
```diff
@@ -102,12 +99,14 @@ public void doRun(ProgressIndicator indicator) {
       return;
     }
 
-    final AtomicReference<Try<@Nullable ExecutionResult>> wrapper = new AtomicReference<>(
+    AtomicReference<Try<@Nullable ExecutionResult>> wrapper = new AtomicReference<>(
         Try.success(null));
+
+    val preRebaseHookExecutor = new PreRebaseHookExecutor(gitRepository);
```
Comment (mkondratek) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/RebaseOnParentBackgroundable.java:105]: o!

File: backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/StatusBranchHookExecutor.java [hunk 3]
```diff
@@ -39,7 +40,10 @@ public final class StatusBranchHookExecutor extends BaseHookExecutor {
 
   @UIThreadUnsafe
   public StatusBranchHookExecutor(IGitCoreRepository gitCoreRepository) {
-    super("machete-status-branch", gitCoreRepository);
+    super("machete-status-branch",
```
Comment (mkondratek) [inline backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/StatusBranchHookExecutor.java:43]: 👍

File: src/test/java/com/virtuslab/archunit/ForbiddenClassesTestSuite.java [hunk 4]
```diff
@@ -10,6 +10,7 @@ public class ForbiddenClassesTestSuite extends BaseArchUnitTestSuite {
   public void no_classes_should_depend_on_java_collections() {
     noClasses()
         .that().resideOutsideOfPackages(
+            "com.virtuslab.gitmachete.backend.hooks",
             "com.virtuslab.gitmachete.frontend.actions..",
```
Comment (mkondratek) [inline src/test/java/com/virtuslab/archunit/ForbiddenClassesTestSuite.java:14]: why are there two subsequent dots `..` sometimes? 🤔

**Review by mkondratek — APPROVED** [review]

File: src/test/java/com/virtuslab/archunit/ForbiddenClassesTestSuite.java (hunk 4 above)
Comment (PawelLipski) [reply src/test/java/com/virtuslab/archunit/ForbiddenClassesTestSuite.java:14]: `foo.bar..` is `foo.bar` and all its subpackages 📦

**Pull request merged**